capsules/src/fxos8700_cq.rs

```diff
@@ -209,20 +220,31 @@ impl<'a, A: time::Alarm + 'a> time::Client for Fxos8700cq<'a, A> {
     }
 }
 
-impl<'a, A: time::Alarm + 'a> I2CClient for Fxos8700cq<'a, A> {
+impl<'a> I2CClient for Fxos8700cq<'a> {
     fn command_complete(&self, buffer: &'static mut [u8], _error: Error) {
         match self.state.get() {
-            State::ReadAccelDelay => {
-                // Set a timer to wait for the conversion to be done.
-                // For 800 Hz, thats 2.25 ms (1/800 Hz + 1)
-                let interval = (2250 as u32) * <A::Frequency>::frequency() / 1000000;
-                let tics = self.alarm.now().wrapping_add(interval);
-                self.alarm.set_alarm(tics);
+            State::ReadAccelSetup => {
+                // Enable the acceleromter.
+                buffer[0] = Registers::CtrlReg1 as u8;
+                buffer[1] = 1;
+                self.i2c.write(buffer, 2);
+                self.state.set(State::ReadAccelWait);
+            }
+            State::ReadAccelWait => {
```

Comment:
Is there a race condition here? Since this is an edge triggered interrupt, if it falls between checking and enabling, we'd miss it?

Why not enable the interrupt before / when sending the command an always wait for the interrupt?